File: 05_ODE/Euler_method/Euler_method.c
#include <stdio.h>
#include <math.h>

#define EPS .00000001

double func_f(double);

int main()
{
    double x = 0.0, y = 0.0;
    double h = 0.01, dx = 1.0, xmax = 10.0; 
    double ddx = 0.0;

    printf("   X\t   Y\n");

    do 
    {
        if(x >= ddx - EPS)
        {
            ddx += dx;
            printf("%7.4f %7.4f\n", x, y);
        }

        y += h * func_f(x);
        x += h;
    }while(x <= xmax);

    return 0;
}

double func_f(double x)
{
    return 2.0 * x;
}

File: 05_ODE/Euler_method/Euler_method.py
# -*- coding: utf-8 -*-

EPS = 0.00000001

x = 0.0
y = 0.0
h = 0.01
dx = 1.0
xmax = 10.0 
ddx = 0.0

def func_f(x):  
    return 2.0 * x

print "   X\t\t   Y"

while True:
    if x >= (ddx - EPS):
        ddx += dx
        print '%7.4f' % x,
        print '\t' + '%7.4f' % y
    
    y += h * func_f(x)
    x += h

    if x > xmax:
        break
